### PR TITLE
Assignments week 1 Oleksandr Starshynov

### PR DESCRIPTION
**Pull request by aleksandrstarshynov**
Updated weather app, week 2, Node module

### CODE REVIEW

File: hackyourtemperature/app.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${keys.API_KEY}&units=metric`);
+    if (!response.ok) {
+      return res.status(response.status).json({error: "API error"});
+    }
+    const data = await response.json();
+    return res.status(200).send(`It is ${data.main.temp}°C in ${cityName}`)
+  } 
+  catch {
+    console.error("Error fetching data");
+    res.status(500).json({ error: "Internal server error" });
```
Comment (EdwardSmart98) [inline hackyourtemperature/app.js:27]: This error is lacking in information the front-end can accurately use, is the error because the API call failed? or is the name of the city not a real city? 
You can check the response from the weather API, and if it's 404, return an error with that info, so the front-end can display this to the user and the user will type a different name

File: hackyourtemperature/sources/keys.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const keys = {API_KEY: "c34f56e7a569f2c2cf3e24aceac5ddc8"};
```
Comment (EdwardSmart98) [inline hackyourtemperature/sources/keys.js:1]: It's good practice to not push up API keys to repo's to keep them safe. two ways to do this are:
-> Ignoring this file in the .gitignore.
-> adding a .env file (and ignoring it) - This is better, as the code will still compile, and the .env file is a standard across a lot of software/programs, so loading them into the program at runtime is handleded well.

**Review by EdwardSmart98 — COMMENTED** [review]
A Strong attempt for the solution, a very strong test suite aswell to go along covering a lot of edge cases. 
The biggest area for improvement is in the error handling, you should try to give consistent and dexript errors to the front-end so they can understand why the error occurred.

**Review by EdwardSmart98 — APPROVED** [review]
